test/rsa_test.c

```diff
@@ -276,13 +299,14 @@ static int test_rsa_oaep(int idx)
     unsigned char ctext[256];
     static unsigned char ptext_ex[] = "\x54\x85\x9b\x34\x2c\x49\xea\x2a";
     unsigned char ctext_ex[256];
+    unsigned char hash_ex[16], hash2_ex[16];
```

Comment:
I think you could avoid these `hash_ex` in these test_rsa_xxx functions by adding a NULL checking in the `SetKey` macro. It would be something like:

```
+    if (h != NULL && h2 != NULL) {          \
+        memcpy(h, hash_ex, sizeof(hash_ex) - 1);                    \
+        memcpy(h2, hash2_ex, sizeof(hash2_ex) - 1);                 \
+    } \
```

And then you can just call the `rsa_setkey` functions as `clen = rsa_setkey(&key, ctext_ex, NULL, NULL, idx);`